### PR TITLE
Simplify & standardise Vagrantfile

### PR DESCRIPTION
**Pull request by edwardmlyte**
Remove non-standard env config, as well as removed Virtualbox and vagrant
checks. Instead reverts back to using environment variables to customize
the VM.
Documentation updated to reflect these changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,29 +12,37 @@ This will install:
...
+## Customising the hardware requirements
+The following environment variables can be set to tune the development environment.
+
+| Variable | What for | Default |
+| -------- | -------- | ------- |
+| `DEVENV_PROCESSORS` | How many CPU cores to allocate | 2 |
+| `DEVENV_CPUEXECUTIONCAP` | The amount of time (as %) a host CPU spends to emulate a virtual CPU | 100 |
+| `DEVENV_MEMORY` | How much memory, in MB, to allocate | 4096 |
+
+## Known Issues
```
Comment (edwardmlyte) [inline README.md:44]: Wanted to really highlight this is not OUR problem. Hence own section

File: Vagrantfile [hunk 2]
```diff
@@ -181,12 +70,12 @@ Vagrant.configure(2) do |config|
     v.customize ["modifyvm", :id, "--cpus", ENV['DEVENV_PROCESSORS'] || "2"]
     v.customize ["modifyvm", :id, "--cpuexecutioncap", ENV['DEVENV_CPUEXECUTIONCAP'] || "100"]
     v.customize ["modifyvm", :id, "--monitorcount", "1"]
-    v.customize ["modifyvm", :id, "--memory", ENV['DEVENV_MEMORY'] || "2048"]
+    v.customize ["modifyvm", :id, "--memory", ENV['DEVENV_MEMORY'] || "4096"]
```
Comment (edwardmlyte) [inline Vagrantfile:73]: Bumped this up to match the newer value we used for the .env file.

**Review by stuart-c-moore — APPROVED** [review]

File: Vagrantfile [hunk 3]
```diff
@@ -148,31 +50,18 @@ Vagrant.configure(2) do |config|
...
-    # downloaded artifacts; to benefit you need to install the plugin on your
-    # host macine by running: vagrant plugin install vagrant-cachier
-    # For more details see: https://github.com/fgrehm/vagrant-cachier
+    # The vagrant-cachier plugin (optional) will speed up rebuilds by reusing downloaded artifacts
     # Configure cached packages to be shared between instances of the same base box.
     config.cache.scope = :box
   end
 
-  # bash history folder needs creating before it is set in the provisioning script below
-  if $windows
```
Comment (edwardmlyte) [inline Vagrantfile:160]: The external bash history folder config feels too custom to roll out to the wider world. Our env shouldn't be creating extra directories really. Maybe could be introduced and moved somewhere more appropriate. Or at the very least mentioned in the README.

**Review by edwardmlyte — COMMENTED** [review]
This essentially removes the internal dev-env's additions that were added. Be interested to get some feedback on this approach.

**Comment by anouskastreets** [comment]
I would like the warning about not logging in before provisioning has completed to stay in the readme. The warning that appears at the start is long gone by the time the login window pops up. If I'm a noob I wouldn't know I can't login.

File: Vagrantfile [hunk 4]
```diff
@@ -181,12 +70,12 @@ Vagrant.configure(2) do |config|
...
     v.customize ["modifyvm", :id, "--cpuexecutioncap", ENV['DEVENV_CPUEXECUTIONCAP'] || "100"]
     v.customize ["modifyvm", :id, "--monitorcount", "1"]
-    v.customize ["modifyvm", :id, "--memory", ENV['DEVENV_MEMORY'] || "2048"]
+    v.customize ["modifyvm", :id, "--memory", ENV['DEVENV_MEMORY'] || "4096"]
     v.customize ["modifyvm", :id, "--vram", "128"]
     v.customize ["modifyvm", :id, "--ioapic", "on"]
     v.customize ["modifyvm", :id, "--accelerate3d", "on"]
     v.customize ["modifyvm", :id, "--clipboard", "bidirectional"]
-    v.customize ["guestproperty", "set", :id, "/VirtualBox/GuestAdd/VBoxService/ — timesync-set-threshold", 10000]
+    v.customize ["guestproperty", "set", :id, "/VirtualBox/GuestAdd/VBoxService/ - timesync-set-threshold", 10000]
```
Comment (stuart-c-moore) [inline Vagrantfile]: I think something may have converted a -- to a - here?

**Comment by edwardmlyte** [comment]
@anouskastreets as far as i can tell, the warning you're referring to in the readme is still there. Straight after the 2 point startup plan.
@migs I noticed that a weird mac-esqu hyphen had been put in instead of a normal one (which resolves to ampersand and some number on non-mac machines), so figured i'd fix that whilst i was digging around. i can revert that if you reckon it's wrong.

File: README.md [hunk 5]
```diff
@@ -12,29 +12,37 @@ This will install:
...
 1. Clone this repository with:
   ```git clone https://github.com/FINkit/development-environment```
 2. Change into the `development-environment` directory and run:
   ```vagrant up```
-3. Follow on screen instructions.
-
-You will be asked a series of questions about what configuration you want to give your development environment. These are stored in a local `.env` file, which can be edited later (or deleted if you want re-prompting).
 
 ## IMPORTANT	
 When the UI loads you will be prompted for a password. Please wait for the entire provisioning process to complete before entering the password. You will be notified in the terminal window when you can login.
```
Comment (edwardmlyte) [inline README.md:31]: @anouskastreets this still ok

**Review by cagiti — APPROVED** [review]

**Pull request merged**